flink-runtime/src/main/java/org/apache/flink/runtime/util/config/memory/CommonProcessMemorySpec.java

```diff
@@ -94,4 +96,16 @@ public MemorySize getTotalFlinkMemorySize() {
 	public MemorySize getTotalProcessMemorySize() {
 		return flinkMemory.getTotalFlinkMemorySize().add(getJvmMetaspaceSize()).add(getJvmOverheadSize());
 	}
+
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
For what do we need the `equals` method on `CommonProcessMemorySpec`?

Reply:
If it is for the tests only, then one could also think about providing a Hamcrest matcher for this task. On the other side, this class is a value class with a well-defined equality relation.

Reply:
Yes, I added `equals` to the memory spec classes because I need them in `ActiveResourceManagerTest`.

> this class is a value class with a well-defined equality relation
Agree, that's way I'm in favor of adding `equals` rather than custom Hamcrest machers.

I'll add the `hashCode` for these classes.